### ISSUE TITLE
docker info shows incorrect info about Total data space

### ISSUE THREAD

**Issue by milindchawre** [moby/moby#27409]
**Description**

Docker info shows incorrect info about Total data space.
Data Space Total != (Data Space Used + Data Space Available)
The instance on which docker is running don't have that much about of space shown in 'Data Space Total' of 'docker info'.

**Steps to reproduce the issue:**
1. docker info
2. Check whether Data Space Total == (Data Space Used + Data Space Available)

**Describe the results you received:**
docker info

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 97
Server Version: 1.12.1
Storage Driver: devicemapper
 Pool Name: docker-253:1-650320-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 5.223 GB
 Data Space Total: 107.4 GB
 Data Space Available: 44.74 GB
 Metadata Space Used: 6.988 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.14 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null overlay host bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.13.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.954 GiB
Name: dockerhost.cisco.com
ID: BLND:CG2Q:6TZJ:KBUU:LPON:F6CG:LZBO:OVKT:P67J:YHQO:XD7M:HE77
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

Space shown in 'Data Space Total' (107.4 GB) is not even available on my instance which is Micro-small flavor (50GB disk space).

**Describe the results you expected:**
docker info should show Data Space Total == (Data Space Used + Data Space Available)
Space shown in 'Data Space Total' should actual exists in reality. For my setup it could be around 50GB (<=50GB) since my instance is Micro-small in flavor with 50GB disk space.

**Output of `docker version`:**

```
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        
 OS/Arch:      linux/amd64
```

**Comment by thaJeztah** [comment]
This is actually because the devicemapper thin pool was initialized as 100GB. Moving that device to a different physical location would allow it to grow to 100GB.

Perhaps this should be documented better in the devicemapper section https://docs.docker.com/engine/userguide/storagedriver/device-mapper-driver/

**Comment by milindchawre** [comment]
I think it would be better if we have this in docker info : 
```
Data Space Used: 5.223 GB
Actual Data Space Total: 50 GB
Data Space Available: 44.74 GB
Data Space Total: 107.4 GB (This is max devicemapper thin pool size which can be extended further)
```

instead of this : 
```
Data Space Used: 5.223 GB
Data Space Total: 107.4 GB
Data Space Available: 44.74 GB
```

So that there is no any confusion and easy to understand.
@thaJeztah ?

**Comment by thaJeztah** [comment]
Perhaps @rhvgoyal has some suggestions/thoughts as well

**Comment by milindchawre** [comment]
@thaJeztah @rhvgoyal  Any updates?

**Comment by foxwang** [comment]
me to
 Data Space Used: 9.024 GB
 Data Space Total: 214.7 GB
 Data Space Available: 15.35 GB

**Comment by thaJeztah** [comment]
The devicemapper storage-driver was deprecated, and has been removed, so closing issues related to this storage-driver. See;

- https://github.com/moby/moby/pull/38017
- https://github.com/moby/moby/pull/43637
